http/src/main/java/org/http4s/blaze/http/parser/BodyAndHeaderParser.java

```diff
@@ -386,6 +387,7 @@ else if (ch == HttpTokens.LF) {
                     final int buff_size = in.remaining();
 
                     if (remaining_chunk_size <= buff_size) {
+                        // End of this chunk
```

Comment:
It could certainly use more docs: do you want that as part of this PR? As far as converting Java to Scala: the performance difference is not huge, but meaningful, and in many ways the state machine is much easier to deal with in Java since we have break etc. Is there a benefit from moving it from Java to Scala? We can entertain rewriting it in a different change if you like.